framework/src/org/checkerframework/framework/type/TypeFromExpressionVisitor.java

```diff
@@ -195,6 +194,11 @@ public AnnotatedTypeMirror visitMemberSelect(MemberSelectTree node,
             if (t instanceof AnnotatedDeclaredType || t instanceof AnnotatedArrayType || t instanceof AnnotatedTypeVariable) {
                 return AnnotatedTypes.asMemberOf(f.types, f, t, elt).asUse();
             }
+        } else {
+            if(node.getIdentifier().contentEquals("class")){
+                // Handle class literals for primitive types (as in int.class)
+                return f.getAnnotatedType(elt);
```

Comment:
This were helpful questions.  As you are pointing out, my statement in the pull request description was wrong

>Primitive class literals were annotated using the default qualifier (the annotation with @DefaultQualifierInHierarchy or whatever DefaultLocation.OTHERWISE) rather than whatever defaulting is applied to fields in class file

The problem isn't the defaulting, but rather that "primitive.class" is annotated via AnnotatedTypeFactory.annotateImplicit(Tree, AnnotatedTypeMirror).  With this change, they are annotated via AnnotatedTypeFactory.annotateImplicit(Element, AnnotatedTypeMirror).  [The inference framework expects types from bytecode to be annotated via annotateImplicit(Element,ATM). ](https://github.com/typetools/checker-framework-inference/blob/master/src/checkers/inference/InferenceAnnotatedTypeFactory.java#L559)

I'm looking now to see if there is a better way to ensure that "primitive.class" is annotated via the .annotateImplicit(Element, AnnotatedTypeMirror) than what I've done here.